node/pkg/driver/node.go

```diff
@@ -613,9 +627,104 @@ func (d *NodeService) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpu
 }
 
 func (d *NodeService) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
-	goid_info.SetAdditionalIDInfo(req.VolumeId)
+	volumeId := req.VolumeId
+	goid_info.SetAdditionalIDInfo(volumeId)
 	defer goid_info.DeleteAdditionalIDInfo()
-	return nil, status.Error(codes.Unimplemented, "NodeGetVolumeStats is not implemented yet")
+	volumePath := req.VolumePath
+	volumePathWithHostPrefix := d.NodeUtils.GetPodPath(req.VolumePath)
+	isFSVolume := true
+
+	err := d.nodeGetVolumeStatsRequestValidation(volumeId, volumePath)
+	if err != nil {
+		return nil, err
+	}
+
+	isPathExists := d.NodeUtils.IsPathExists(volumePathWithHostPrefix)
+	if !isPathExists {
+		return nil, status.Errorf(codes.NotFound, "volume path %q does not exists", volumePath)
+	}
+	isBlock, err := d.NodeUtils.isBlock(volumePathWithHostPrefix)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to determine if %q is block device: %s", volumePath, err)
+	}
+	if isBlock {
+		isFSVolume = false
+	}
+	isMounted, err := d.IsMounted(volumePathWithHostPrefix, isFSVolume)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to check if volume path %q is mounted", volumePath)
+	}
+	if !isMounted {
+		return nil, status.Errorf(codes.NotFound, "volume path %q is not mounted", volumePath)
+	}
+	volumeStats, err := d.getVolumeStats(volumePathWithHostPrefix)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to get capacity statistics for volume path %q: %s", volumePath, err)
+	}
+	return &csi.NodeGetVolumeStatsResponse{
+		Usage: []*csi.VolumeUsage{
+			{
+				Unit:      csi.VolumeUsage_BYTES,
+				Available: volumeStats.availableBytes,
+				Total:     volumeStats.totalBytes,
+				Used:      volumeStats.usedBytes,
+			},
+			{
+				Unit:      csi.VolumeUsage_INODES,
+				Available: volumeStats.availableInodes,
+				Total:     volumeStats.totalInodes,
+				Used:      volumeStats.usedInodes,
+			},
+		},
+	}, nil
+}
+
+func (d *NodeService) nodeGetVolumeStatsRequestValidation(volumeId string, volumePath string) error {
+	if volumeId == "" {
+		return status.Error(codes.InvalidArgument, "NodeGetVolumeStats Volume ID must be provided")
+	}
+	if volumePath == "" {
+		return status.Error(codes.InvalidArgument, "NodeGetVolumeStats Volume Path must be provided")
+	}
+
+	return nil
+}
+
+//func (d *NodeService) isBlock(devicePath string) (bool, error) {
+//	var stat unix.Stat_t
+//	err := unix.Stat(devicePath, &stat)
+//	if err != nil {
+//		return false, err
+//	}
+//	return (stat.Mode & unix.S_IFMT) == unix.S_IFBLK, nil
+//}
```

Comment:
if you want to experiment, please move this PR to draft until you finish (we don't include commented-out code in PRs)

Reply:
sorry it was for Arbel to see something, removed it